git-path/src/realpath.rs

```diff
@@ -46,8 +46,8 @@ pub(crate) mod function {
                     }
                     CurDir => traverse(input_path, num_symlinks, max_symlinks, real_path),
                     ParentDir => {
-                        let parent_path = PathBuf::from(real_path.parent().ok_or(Error::MissingParent {
-                            path: real_path.to_string_lossy().into(),
+                        let parent_path = PathBuf::from(real_path.parent().ok_or_else(|| Error::MissingParent {
```

Comment:
why is this better?

Reply:
Maybe I am not fully understanding the question as it's made on code that I don't think exists like that anymore. 
But just looking at the diff in the main page then the answer is:

- `to_string_lossy()` should always be avoided in favor of the original data that isn't degenerated. The path isn't only for printing, but callers may also use it themselves.
- `ok_or_else()` is something that clippy wants due to the function call.